### PR TITLE
Adds support for SSL with signed certificates to Builder-Minio package

### PR DESCRIPTION
**Pull request by eeyun**
Signed-off-by: Ian Henry <ihenry@chef.io>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/builder-minio/habitat/hooks/init [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+exec 2>&1
+
+{{#if cfg.use_ssl}}
+    mkdir -p {{pkg.svc_config_path}}/certs
+    ln -s {{pkg.svc_files_path}}/private.key {{pkg.svc_config_path}}/certs/private.key
```
Comment (chefsalim) [inline components/builder-minio/habitat/hooks/init]: Should the key file names be configurable?

Comment (eeyun) [reply components/builder-minio/habitat/hooks/init]: They should be, but minio doesn't support that, they have to be named this specific way.

File: components/builder-minio/habitat/hooks/init [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+exec 2>&1
+
+{{#if cfg.use_ssl}}
+    mkdir -p {{pkg.svc_config_path}}/certs
+    cp {{pkg.svc_files_path}}/private.key {{pkg.svc_config_path}}/certs/private.key
```
Comment (eeyun) [inline components/builder-minio/habitat/hooks/init]: By switching these to `cp` from `ln` we'll get a loop on init failure until the files get uploaded to the service group which is exactly what we want here.

Comment (chefsalim) [reply components/builder-minio/habitat/hooks/init]: 👍

**Pull request merged**